lib/cylc/rundb.py

```diff
@@ -486,14 +503,27 @@ def select_broadcast_states(self, callback, id_key=None):
         select from broadcast_states table if id_key == CHECKPOINT_LATEST_ID.
         Otherwise select from broadcast_states_checkpoints where id == id_key.
         """
+        stmt, stmt_args = self.pre_select_broadcast_states(id_key=None,
+                                                           order=sort)
+        for row_idx, row in enumerate(self.connect().execute(stmt, stmt_args)):
+            callback(row_idx, list(row))
+
+    def pre_select_broadcast_events(self, order=None):
```

Comment:
Good point, though I assume in this & the other case with an identical comment that you mean ``@classmethod``, since class and not instance variables are referenced, & imply decorating as such after changes to attribute referencing (``self.TABLE_BROADCAST_EVENTS`` can & indeed should be ``cls.TABLE_BROADCAST_EVENTS``). Addressing in my next commit.